### DISCUSSION TITLE
Use expressions to add values

### DISCUSSION THREAD

**Discussion by d-sooter** [gorules/zen#197] [Q&A] [answered]
We have large payloads and would like to use the expression node to add calculated fields without having to manually add all the fields.

I wasn’t able to find a way to do it in the docs. As far as i can tell the payload is completely replaced. 

Also it would be awesome if the result of the first expression was passed to the second one that way it could be used for further calculations. 

Maybe it works and im just to daft

**Comment by ivanmiletic** [comment]
Hi, Expression node allows to set values to be used in the scope of the expression node, it is not yet documented but will be in upcoming few weeks. To do it you can access previously calculated value with $.<key> example

![Screenshot 2024-07-01 at 14 23 05](https://github.com/gorules/zen/assets/60513195/22b1ec36-90c3-4ab5-a615-8377ef05c376)

**Comment by nimbit-software** [comment]
Awesome thanks for the quick response. What about keeping the existing payload? Would that work too. 

lets say i want to just add a calculated column (colC) and it is colA * colB

Payload
````
{
colA: 1,
colB: 2
}
````

My goal is to add to existing payload using an expression

**Comment by ivanmiletic** [comment, marked as answer]
That works out of box, downstream nodes will just merge outputs from connected nodes:

![Screenshot 2024-07-01 at 14 49 09](https://github.com/gorules/zen/assets/60513195/4131c73a-54c0-4067-8f06-806f6b385898)
![Screenshot 2024-07-01 at 14 49 22](https://github.com/gorules/zen/assets/60513195/fbe255b5-6cd7-4ee3-8879-c49d5e32339c)

**Comment by nimbit-software** [comment]
Awesome, and once again thanks for the quick response

**Comment by ivanmiletic** [comment]
Will mark this as resolved and move to discussion
